lib/vnet/install_service_windows.go

```diff
@@ -0,0 +1,168 @@
+// executable at that path, we use a path under %PROGRAMFILES%, which is not
+// writable by regular users by default.
+func InstallService(ctx context.Context) (err error) {
+	tshPath, err := os.Executable()
+	if err != nil {
+		return trace.Wrap(err, "getting current exe path")
+	}
+	if err := assertProgramFiles(tshPath); err != nil {
+		return trace.Wrap(err, "checking if tsh.exe is installed under %%PROGRAMFILES%%")
+	}
+	if err := assertWintunInstalled(tshPath); err != nil {
+		return trace.Wrap(err, "checking if wintun.dll is installed next to %s", tshPath)
+	}
+
+	svcMgr, err := mgr.Connect()
+	if err != nil {
+		return trace.Wrap(err, "connecting to Windows service manager")
+	}
+	svc, err := svcMgr.OpenService(serviceName)
+	if err != nil {
+		// The service has not been created yet and must be installed.
+		svc, err = svcMgr.CreateService(
+			serviceName,
+			tshPath,
+			mgr.Config{
+				StartType: mgr.StartManual,
+			},
+			ServiceCommand,
+		)
+		if err != nil {
+			return trace.Wrap(err, "creating VNet Windows service")
+		}
+	}
+	if err := svc.Close(); err != nil {
+		return trace.Wrap(err, "closing VNet Windows service")
+	}
+	if err := grantServiceRights(); err != nil {
+		return trace.Wrap(err, "granting authenticated users permission to control the VNet Windows service")
+	}
+	return nil
+}
+
+// UninstallService uninstalls the VNet windows service.
+func UninstallService(ctx context.Context) (err error) {
+	svcMgr, err := mgr.Connect()
+	if err != nil {
+		return trace.Wrap(err, "connecting to Windows service manager")
+	}
+	svc, err := svcMgr.OpenService(serviceName)
+	if err != nil {
+		return trace.Wrap(err, "opening Windows service %s", serviceName)
+	}
+	if err := svc.Delete(); err != nil {
+		return trace.Wrap(err, "deleting Windows service %s", serviceName)
+	}
+	if err := svc.Close(); err != nil {
+		return trace.Wrap(err, "closing VNet Windows service")
+	}
+	return nil
+}
+
+func grantServiceRights() error {
+	// Get the current security info for the service, requesting only the DACL
+	// (discretionary access control list).
+	si, err := windows.GetNamedSecurityInfo(serviceName, windows.SE_SERVICE, windows.DACL_SECURITY_INFORMATION)
+	if err != nil {
+		return trace.Wrap(err, "getting current service security information")
+	}
+	// Get the DACL from the security info.
+	dacl, _ /*defaulted*/, err := si.DACL()
+	if err != nil {
+		return trace.Wrap(err, "getting current service DACL")
+	}
+	// Build an explicit access entry allowing authenticated users to start,
+	// stop, and query the service.
+	ea := []windows.EXPLICIT_ACCESS{{
+		AccessPermissions: windows.SERVICE_QUERY_STATUS | windows.SERVICE_START | windows.SERVICE_STOP,
+		AccessMode:        windows.GRANT_ACCESS,
+		Trustee: windows.TRUSTEE{
+			TrusteeForm:  windows.TRUSTEE_IS_NAME,
+			TrusteeType:  windows.TRUSTEE_IS_WELL_KNOWN_GROUP,
+			TrusteeValue: windows.TrusteeValueFromString("Authenticated Users"),
```

Comment:
vnet tunnels are specific to each user (even if configured for the entire machine), right? If so, is there any meaningful security impact in one user starting/stopping/querying the service, if it is in use by another user?

Reply:
I suppose it would be similar to another user being able to disconnect the machine from a VPN, or connecting to a different VPN. So yes, there's some impact to other users being able to start and stop VNet, but I'm not sure there's much we can do about that with a per-machine installation. The VNet security invariants are not really ideal for multi-user machines in the first place, given that any user could make a TCP connection to one of your Teleport apps authenticated as the active VNet user. But again, this is similar to a VPN

Reply:
Not a blocker for this PR, but because of this we may want to add the ability to set the group allowed to start/stop the service at some point.